### PR TITLE
Fix test fetcher (doctest) + `Idefics2`'s doc example

### PR DESCRIPTION
**Pull request by ydshieh**
# What does this PR do?

- test fetcher (doctest): a mistake I made in the previous PR
- `Idefics2`'s doc example: not able to fit into GPU even with fp16. So I changed it.

### CODE REVIEW

File: utils/tests_fetcher.py [hunk 1]
```diff
@@ -507,7 +507,7 @@ def get_all_doctest_files() -> List[str]:
     # change to use "/" as path separator
     test_files_to_run = ["/".join(Path(x).parts) for x in test_files_to_run]
     # don't run doctest for files in `src/transformers/models/deprecated`
-    test_files_to_run = [x for x in test_files_to_run if "models/deprecated" not in test_files_to_run]
+    test_files_to_run = [x for x in test_files_to_run if "models/deprecated" not in x]
```
Comment (ydshieh) [inline utils/tests_fetcher.py:510]: my bad

File: src/transformers/models/idefics2/modeling_idefics2.py [hunk 2]
```diff
@@ -1805,15 +1801,16 @@ def forward(
...
         ...   "In which city is that bridge located?<image>",
-        >>> ]
+        ... ]
         >>> images = [[image1, image2], [image3]]
-        >>> inputs = processor(text=prompts, padding=True, return_tensors="pt").to(DEVICE)
+        >>> inputs = processor(text=prompts, padding=True, return_tensors="pt").to("cuda")
 
         >>> # Generate
-        >>> generated_ids = model.generate(**inputs, bad_words_ids=BAD_WORDS_IDS, max_new_tokens=500)
+        >>> generated_ids = model.generate(**inputs, bad_words_ids=BAD_WORDS_IDS, max_new_tokens=20)
```
Comment (ydshieh) [inline src/transformers/models/idefics2/modeling_idefics2.py:1809]: it's very slow, so let's just use `20`

File: src/transformers/models/idefics2/modeling_idefics2.py [hunk 3]
```diff
@@ -1805,15 +1801,16 @@ def forward(
...
         >>> generated_texts = processor.batch_decode(generated_ids, skip_special_tokens=True)
 
         >>> print(generated_texts)
+        ['In this image, we can see the city of New York, and more specifically the Statue of Liberty. In this image, we can see the city of New York, and more specifically the Statue of Liberty.\n\n', 'In which city is that bridge located?\n\nThe bridge is located in the city of Pittsburgh, Pennsylvania.\n\n\nThe bridge is']
```
Comment (ydshieh) [inline src/transformers/models/idefics2/modeling_idefics2.py:1813]: The results are not very good. But we don't have it previously, so I don't know if we consider it bad.

Comment (amyeroberts) [reply src/transformers/models/idefics2/modeling_idefics2.py:1813]: I'm surprised by these generations - they also don't look like the outputs when I was integrating the model 🤔

**Review by amyeroberts — APPROVED** [review]
Thanks for handling this!

File: src/transformers/models/idefics2/modeling_idefics2.py (hunk 3 above)
Comment (ydshieh) [reply src/transformers/models/idefics2/modeling_idefics2.py:1813]: maybe you were using other type of GPUs. CI is using T4.

Comment (amyeroberts) [reply src/transformers/models/idefics2/modeling_idefics2.py:1813]: Quite possibly - I was using `NVIDIA A10G`. I find it quite concerning the generations can be so much worse between hardware though, especially given this isn't a tiny model

Comment (ydshieh) [reply src/transformers/models/idefics2/modeling_idefics2.py:1813]: I can check it tomorrow. But I will merge first 🙏

**Pull request merged**